packages/grafana-ui/src/components/uPlot/utils.ts

```diff
@@ -154,7 +154,7 @@ export function preparePlotData2(
   stackingGroups: StackingGroup[],
   onStackMeta?: (meta: StackMeta) => void
 ) {
-  let data: AlignedData = Array(frame.fields.length);
+  let data = Array(frame.fields.length);
```

Comment:
Instead of removing the type I think I could also replace it by `[]` and remove `= Array(frame.fields.length)`.

Reply:
[Looks like](https://github.com/grafana/grafana/pull/84470#issuecomment-2020635477) now the type of `data` is `any` in the return value of `getDerivedStateFromProps()` though? 🤔

Reply:
Yes, this was actually my initial plan. 
Since introducing `any` isn't really a good solution and defining the length of the array with `Array(frame.fields.length)` doesn't seem necessary I came up with another option right above your comment, @leventebalogh. What do you think?

Reply:
@leventebalogh ?

Reply:
@L-M-K-B the point leve is making is that by removing this type you cause a breaking change in the return types because `data` is no longer typed (see https://github.com/grafana/grafana/pull/84470#issuecomment-2020635477)

believe you can fix this by just adding a return type to the `preparePlotData2` function, so:
```tsx
export function preparePlotData2(
  frame: DataFrame,
  stackingGroups: StackingGroup[],
  onStackMeta?: (meta: StackMeta) => void
): AlignedData {
```

Reply:
@ashharrison90 Unfortunately not. 
<img width="967" alt="Screenshot 2024-04-12 at 12 35 17" src="https://github.com/grafana/grafana/assets/48948963/d5abca02-2a0f-400f-8633-ae17b071d0e4">

Reply:
@L-M-K-B you don't need the type annotation on `data`, just on the `preparePlotData` function